backend/src/db/migrations/20240810015050_consumer-secret.ts

```diff
@@ -0,0 +1,68 @@
+import { Knex } from "knex";
+
+import { SecretEncryptionAlgo, SecretKeyEncoding, TableName } from "@app/db/schemas";
+import { createOnUpdateTrigger, dropOnUpdateTrigger } from "@app/db/utils";
+
+export async function up(knex: Knex): Promise<void> {
+  // TODO: blind index and versioning
+
+  if (!(await knex.schema.hasTable(TableName.ConsumerSecret))) {
+    await knex.schema.createTable(TableName.ConsumerSecret, (t) => {
+      t.uuid("id", { primaryKey: true }).defaultTo(knex.fn.uuid());
+      // t.integer("version").defaultTo(1).notNullable();
+      t.string("type").notNullable(); // login, cc, not, etc.
+      t.text("secretNameCiphertext").notNullable();
+      t.text("secretNameIV").notNullable();
+      t.text("secretNameTag").notNullable();
+      // different types require a different amount of fields, currently support up to 4
+      t.text("secretFieldOneCiphertext").notNullable();
```

Comment:
Intresting, how would you make this more dynamic? so that even though we may have 50 different types of creds, we only have lets say one object?